### ISSUE TITLE
differing case for package name confuses poetry

### ISSUE THREAD

**Issue by davidszotten** [python-poetry/poetry#2487]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

 i'm not sure exactly how to describe this issue so apologies if a duplicate issue exists

- **OS: OSX 10.15.4**
- **Poetry: 1.0.5**

## Issue
trying to add the same package with differing case seems to confuse poetry into thinking that they're different causing it to then fail to resolve two different version constraints

```
cd <empty folder>
poetry new .
poetry add django==2  # note all lowercase
# ... installs

$ poetry add Django==3  # note capitalised

Updating dependencies
Resolving dependencies... (0.0s)

[SolverProblemError]
Because poetrytest depends on both django (2) and Django (3), version solving failed.
```

**Comment by danihodovic** [comment]
Experiencing the same issue with Poetry 1.1.5 during an upgrade from Django 3.0 to Django 3.2. 

```  
SolverProblemError

  Because findwork.dev depends on both django (3.0) and Django (3.2), version solving failed.


```

**Comment by mkniewallner** [comment]
This has been fixed in the to-be-released Poetry `1.2` version.

Tried with [1.2.0b2](https://github.com/python-poetry/poetry/releases/tag/1.2.0b2) for instance:
```console
$ poetry new .
Created package foo in .

$ poetry add django==2
Creating virtualenv foo-V-ORBwh6-py3.9 in <REDACTED>

Updating dependencies
Resolving dependencies... (0.5s)

Writing lock file

Package operations: 2 installs, 0 updates, 0 removals

  • Installing pytz (2022.1)
  • Installing django (2.0)

$ poetry add Django==3

Updating dependencies
Resolving dependencies... (0.3s)

Writing lock file

Package operations: 2 installs, 1 update, 0 removals

  • Installing asgiref (3.5.2)
  • Installing sqlparse (0.4.2)
  • Updating django (2.0 -> 3.0)

$ poetry add django==4

Updating dependencies
Resolving dependencies... (0.1s)

Writing lock file

Package operations: 0 installs, 1 update, 1 removal

  • Removing pytz (2022.1)
  • Updating django (3.0 -> 4.0)

$ cat pyproject.toml
...
[tool.poetry.dependencies]
python = "^3.9"
Django = "4"
...
```
